app_dart/lib/src/service/scheduler.dart

```diff
@@ -1091,31 +1153,26 @@ class Scheduler {
     final logCrumb = 'checkCompleted($name, $slug, $sha, $conclusion)';
     final documentName = CiStaging.documentNameFor(slug: slug, sha: sha, stage: stage);
     log.info('$logCrumb: $documentName');
-    StagingConclusion stagingConclusion;
-    try {
-      // We're doing a transactional update, which could fail if multiple tasks are running at the same time; so retry
-      // a sane amount of times before giving up.
-      const RetryOptions r = RetryOptions(
-        maxAttempts: 3,
-        delayFactor: Duration(seconds: 2),
-      );
-      stagingConclusion = await r.retry(
-        () => markCheckRunConclusion(
+
+    // We're doing a transactional update, which could fail if multiple tasks are running at the same time; so retry
+    // a sane amount of times before giving up.
+    const RetryOptions r = RetryOptions(
+      maxAttempts: 3,
+      delayFactor: Duration(seconds: 2),
+    );
+
```

Comment:
What should we do if this throws an error?

Reply:
I thought this was the case of "everything is on fire". Normally, we just let the error propagate and handle it at the request handler level. I thought the only reason we had a catch-all here is because we were cautious about this logic when introducing it into flutter/flutter.